types/index.d.ts

```diff
@@ -101,3 +101,40 @@ export type Pick4<
     };
   };
 };
+
+// https://github.com/Microsoft/TypeScript/pull/21316
+
+/**
+ * Extracts union strings of all property names in T whose value is a function.
+ */
+// tslint:disable-next-line: semicolon strict-export-declare-modifiers ban-types
+export type FunctionPropertyNames<T> = { [K in keyof T]: T[K] extends Function ? K : never; }[keyof T];
+
+/**
+ * Creates a type that only includes the properties which are functions in T.
+ */
+export type FunctionProperties<T> = Pick<T, FunctionPropertyNames<T>>;
+
+/**
+ * Extracts union strings of all property names in T whose value is not a function.
+ */
+// tslint:disable-next-line: semicolon strict-export-declare-modifiers ban-types
+export type NonFunctionPropertyNames<T> = { [K in keyof T]: T[K] extends Function ? never : K; }[keyof T];
+
+/**
+ * Creates a type that only includes the properties which are functions in T.
+ */
+export type NonFunctionProperties<T> = Pick<T, NonFunctionPropertyNames<T>>;
```

Comment:
These don't seem to me to be of general enough utility to define them as top-level exports. I would get rid of `FunctionPropertyNames`/`FunctionProperties` since they aren't needed in the definition of `DeepReadonly`, inline `NonFunctionPropertyNames`/`NonFunctionProperties`.